src/Microsoft.IdentityModel.Protocol.Extensions/Configuration/HttpDocumentRetriever.cs

```diff
@@ -46,17 +49,19 @@ public HttpDocumentRetriever(HttpClient httpClient)
         {
             if (string.IsNullOrWhiteSpace(address))
             {
-                throw new ArgumentNullException("address");
+                LogHelper.Throw(string.Format(CultureInfo.InvariantCulture, ErrorMessages.IDX10000, GetType() + ": address"), typeof(ArgumentNullException), EventLevel.Verbose);
             }
             try
             {
+                IdentityModelEventSource.Logger.WriteVerbose("Obtaining information from metadata endpoint: " + address);
                 HttpResponseMessage response = await _httpClient.GetAsync(address, cancel).ConfigureAwait(false);
                 response.EnsureSuccessStatusCode();
                 return await response.Content.ReadAsStringAsync().ConfigureAwait(false);
             }
             catch (Exception ex)
             {
-                throw new IOException("Unable to get document from: " + address, ex);
+                LogHelper.Throw(string.Format(CultureInfo.InvariantCulture, ErrorMessages.IDX10804, address), typeof(IOException), EventLevel.Error, ex);
+                return null;
```

Comment:
We can design this slightly differently do we don't need the null.  If LogHelper returns an exception, then you could do: throw LogHelper.Throw(...). But that is for later.